dotnet/src/SemanticKernel/Orchestration/SKFunction.cs

```diff
@@ -159,37 +159,35 @@ public FunctionView Describe()
         SKContext? context = null,
         CompleteRequestSettings? settings = null,
         ILogger? log = null,
-        CancellationToken? cancel = null)
+        CancellationToken cancellationToken = default)
     {
         if (context == null)
         {
-            var cToken = cancel ?? default;
             log ??= NullLogger.Instance;
             context = new SKContext(
                 new ContextVariables(""),
                 NullMemory.Instance,
                 this._skillCollection,
                 log,
-                cToken);
+                cancellationToken);
         }
 
         context.Variables.Update(input);
 
-        return this.InvokeAsync(context, settings, log, cancel);
+        return this.InvokeAsync(context, settings, log, cancellationToken);
     }
 
     /// <inheritdoc/>
     public Task<SKContext> InvokeAsync(
```

Comment:
There's some funny, misleading stuff going on in these two InvokeAsync() methods (predates this PR).

If the user passes all the args, 2 of them will be ignored unless one (context) is also null. If the user passes a default (empty) SKContext, thinking nothing of it except that they're filling in the blanks, why would they think their logger or cancellation token would be ignored?

Looks to me like these should each be split into overloads: 
- One with only SKContext (non-nullable), settings, and (for method above) input. 
- One with no log and cancellationToken (and input), and no SKContext.

@stephentoub  if you agree, I'll file a task and have someone make the change.

Reply:
> @stephentoub if you agree

Yup.  I hadn't noticed it for the logger, but this is the crux of the issue I was commenting on for the cancellation token.  There should only be one way to pass in a given piece of information; if there are two, either one gets ignored (which is bad) or both should be used (which is complicated).

So basically if an SKContext is accepted, that same method shouldn't also take anything contained in the SKContext, e.g. the cancellation token.

Reply:
Thanks! I'll take care of these InvokeAsync overloads